### PR TITLE
Add a banner celebrating smallpox eradication day

### PR DESCRIPTION
**Pull request by jpaddison3**
*Never forget what we are capable of when we band together and declare battle on what is broken in the world.*

I actually did it right and used srcset.

### CODE REVIEW

**Comment by jpaddison3** [comment]
To test, set `showSmallpox: true` in your public database settings.

File: packages/lesswrong/components/ea-forum/EAHome.tsx [hunk 1]
```diff
@@ -1,23 +1,28 @@
 import React from 'react'
-import { userHasEAHomeHandbook } from '../../lib/betas'
-import { PublicInstanceSetting } from '../../lib/instanceSettings'
+// import { userHasEAHomeHandbook } from '../../lib/betas'
+// import { PublicInstanceSetting } from '../../lib/instanceSettings'
```
Comment (colophonemes) [inline packages/lesswrong/components/ea-forum/EAHome.tsx:3]: 💀 ?

File: packages/lesswrong/components/ea-forum/EAHome.tsx [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+import { DatabasePublicSetting } from '../../lib/publicSettings'
 import { Components, registerComponent } from '../../lib/vulcan-lib'
 import { useCurrentUser } from '../common/withUser'
 
-const eaHomeSequenceIdSetting = new PublicInstanceSetting<string | null>('eaHomeSequenceId', null, "optional") // Sequence ID for the EAHomeHandbook sequence
+// const eaHomeSequenceIdSetting = new PublicInstanceSetting<string | null>('eaHomeSequenceId', null, "optional") // Sequence ID for the EAHomeHandbook sequence
```
Comment (colophonemes) [inline packages/lesswrong/components/ea-forum/EAHome.tsx:8]: 💀 ?

File: packages/lesswrong/components/ea-forum/EAHome.tsx [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
 const EAHome = () => {
   const currentUser = useCurrentUser();
   const {
-    RecentDiscussionFeed, HomeLatestPosts, EAHomeHandbook, RecommendationsAndCurated
+    RecentDiscussionFeed, HomeLatestPosts, EAHomeHandbook, RecommendationsAndCurated, SmallpoxBanner
   } = Components
 
   const recentDiscussionCommentsPerPost = (currentUser && currentUser.isAdmin) ? 4 : 3;
-  const shouldRenderEAHomeHandbook = userHasEAHomeHandbook(currentUser)
+  // const shouldRenderEAHomeHandbook = userHasEAHomeHandbook(currentUser)
```
Comment (colophonemes) [inline packages/lesswrong/components/ea-forum/EAHome.tsx:18]: OK, I think I get it – you're removing the handbook links for now, and then you'll put this back later. TBH I'm not a huge fan of this, it seems brittle, and it raises the question of what's going to happen if you want to add a different banner at some point in the future. I think I'd prefer this to be guarded, either with a db setting like `showHomeHandbookBanner`, or just a check to see if any other banners are going to be rendered. Then the code is more reusable in future – you can easily update the DB setting to get the banner back if you want to run it again next year, and you can use the same pattern for other banners in the future.

File: packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}))
+
+const SmallpoxBanner = ({ classes }) => {
+  const { SingleColumnSection } = Components
+  
+  return <SingleColumnSection>
+    <Link to={featuredPost} className={classes.link}>
+      <picture>
+        {/* 960px is the md boundary */}
+        <source media="(max-width: 959px)" srcSet={mobileImage} />
```
Comment (colophonemes) [inline packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx:34]: Not sure if your version of material supports this, but if it does, you could also get `theme` object with the [`useTheme()` hook](https://material-ui.com/styles/api/#usetheme-theme), and then use `theme.breakpoints.values.sm` to get the value.

Relatedly (and very minor), you might also want to express it as `960 - 0.05px`/`959.95` to account for devices with 2x resolution that can resolve sub-pixel values.

File: packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        <source media="(min-width: 960px)" srcSet={desktopImage} />
+        <img
+          className={classes.image}
+          src={desktopImage}
+          height={bannerHeight}
```
Comment (colophonemes) [inline packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx:39]: This is kind of a deprecated way to specify image height – just specify it using CSS. You're already applying a style here, so would move to the `createStyles` block.

File: packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+const desktopImage = `https://res.cloudinary.com/${cloudinaryCloudNameSetting.get()}/image/upload/w_${SECTION_WIDTH},h_${bannerHeight}/${desktopImageId}`
+
+const styles = createStyles((theme: ThemeType): JssStyles => ({
+  link: {
+    '&:hover': {
+      opacity: 'unset'
+    }
+  },
+  image: {
+    objectFit: 'cover'
```
Comment (colophonemes) [inline packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx:23]: I'm slightly suspicious of this (I'm not sure why it doesn't break layout on mobile adding a horizontal scrollbar, but I guess there's some ancestor component that has `overflow: hidden` set). Seems OK for now...

**Review by colophonemes — CHANGES_REQUESTED** [review]
I think the dead code isn't great, and I'd suggest refactoring so that you don't need to make a code change to take down the banner.

File: packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx (hunk 6 above)
Comment (jpaddison3) [reply packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx:23]: You think I should add `overflow: hidden` as well to be safe?

**Comment by jpaddison3** [comment]
> I'd suggest refactoring so that you don't need to make a code change to take down the banner.

Which banner do you mean? The smallpox banner can be taken down with a settings update. The Handbook banner I agree.

File: packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx (hunk 4 above)
Comment (jpaddison3) [reply packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx:34]: There's no `useTheme()` on this version of MUI.

File: packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx (hunk 6 above)
Comment (colophonemes) [reply packages/lesswrong/components/ea-forum/SmallpoxBanner.tsx:23]: Yeah, I think it's probably fine, although I have some taste sense that it's not great to have elements falling off the screen in a somewhat uncontrolled way. Not sure what this'll do e.g. in Edge

**Review by colophonemes — APPROVED** [review]
LGTM

**Pull request merged**